### PR TITLE
[COMCTL32] Fix edit UAF

### PR DESCRIPTION
**Pull request by learn-more**
Noticed while running with DPH

### CODE REVIEW

File: dll/win32/comctl32/edit.c [hunk 1]
```diff
@@ -3397,12 +3397,18 @@ static LRESULT EDIT_WM_KeyDown(EDITSTATE *es, INT key)
...
 
     es->flags &= ~EF_FOCUSED;
     DestroyCaret();
     if (!(es->style & ES_NOHIDESEL))
         EDIT_InvalidateText(es, es->selection_start, es->selection_end);
     EDIT_NOTIFY_PARENT(es, EN_KILLFOCUS);
+
+    // Someone destroyed us in the EN_KILLFOCUS handler, so we can't use 'es' anymore
+    if (!IsWindow(hwndSelf))
+        return 0;
```
Comment (binarymaster) [inline dll/win32/comctl32/edit.c:3410]: This is Wine code btw.

**Comment by JoachimHenze** [comment]
Could you please give some simple reproduction steps what to do (beside enabling DPH) to trigger this @learn-more ?
Thank you!

**Comment by learn-more** [comment]
> Could you please give some simple reproduction steps what to do (beside enabling DPH) to trigger this @learn-more ? Thank you!

Start renaming a file, then cancel that.

File: dll/win32/comctl32/edit.c (hunk 1 above)
Comment (learn-more) [reply dll/win32/comctl32/edit.c:3410]: https://gitlab.winehq.org/wine/wine/-/merge_requests/1207

**Review by JoachimHenze — APPROVED** [review]

File: dll/win32/comctl32/edit.c (hunk 1 above)
Comment (binarymaster) [reply dll/win32/comctl32/edit.c:3410]: New patch :smile: https://gitlab.winehq.org/wine/wine/-/merge_requests/1270

**Comment by binarymaster** [comment]
https://gitlab.winehq.org/wine/wine/-/merge_requests/1270 is now merged, you can update this PR as a Winesync.

**Comment by KRosUser2** [comment]
Good news if merged in Wine.
Second approval needed for merge now ?

**Comment by learn-more** [comment]
https://build.reactos.org/#/builders/9/builds/33331

**Review by ColinFinck — APPROVED** [review]
LGTM, especially considering that the very same patch also made its way into WINE.
Despite the promise at https://gitlab.winehq.org/wine/wine/-/merge_requests/1270/diffs#note_13375, there hasn't been a follow-up commit to guard `es` yet. This should be kept in mind and later synced.

However, there have been a few more commits to the same file in WINE, so you probably want to sync them as well in a follow-up PR.

**Comment by JoachimHenze** [comment]
I tested the reproduction steps with a system-wide DPH build GCC4.7.2dbg of 0.4.7-release-234-gdc95ee3 (which is yet unpatched):
Repro-steps: "start renaming a file, then cancel that via ESC". I tried it on a drive, a folder and a file.
None of them showed anything suspicious in the log, everything behaved like I would expect it.

**Pull request merged**